src/services.ts

```diff
@@ -30,33 +32,161 @@ export class ServiceCompletionProvider implements vscode.CompletionItemProvider
                                 }
                             }
                         }
-
                         return true
-                    }))
+                    }),
+                    MemoryCategory: klass.MemoryCategory,
+                    Name: klass.Name,
+                    Superclass: klass.Superclass,
+                    Tags: klass.Tags,
+                }
+                output.set(klass.Name, klassData)
+            }
+
+            return output
+        })
+
+        this.servicesCompletion = getApiDump().then((dump) => {
+            const output: vscode.CompletionItem[] = []
+
+            for (const klass of dump.Classes) {
+                if (klass.Tags !== undefined && klass.Tags.includes("Service")) {
+                    const completionItem = new vscode.CompletionItem(klass.Name, vscode.CompletionItemKind.Class)
+
+                    completionItem.detail = `(service) ${klass.Name}`
+                    completionItem.documentation = new vscode.MarkdownString(`[Developer Reference](https://developer.roblox.com/en-us/api-reference/class/${klass.Name})`)
+
+                    output.push(completionItem)
                 }
             }
 
             return output
         })
     }
 
-    async provideCompletionItems(document: vscode.TextDocument, position: vscode.Position) {
+    public async createCompletionItems(
+        service: ApiClass,
+        operator: string,
+        inheritMembers = true,
```

Comment:
When would this be false?

Reply:
Valid point, I was probably thinking more of a utility function used in multiple cases when I was writing this, but this is clearly just used in this specific class, will remove.
It would probably also make sense in future to start separating out code into common (utility?) modules as there does seem to be some overlapping code

Reply:
If it was a utility function I would probably want to avoid boolean parameters anyway, just because I consider them a code smell. For now it's probably fine to just remove it.